nullaway/src/test/java/com/uber/nullaway/jspecify/ArrayTests.java

```diff
@@ -470,6 +471,51 @@ public void localVariableIndexArray() {
         .doTest();
   }
 
+  @Test
+  public void loopVariableIndex() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "  static @Nullable String[] fizz = {\"1\"};",
+            "  static void foo() {",
+            "    for (int i = 0; i < fizz.length; i++) {",
+            "      if (fizz[i] != null) {",
+            "        fizz[i].toString();",
+            "      }",
+            "    }",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  @Ignore("for-each handling needs to be fixed; see https://github.com/uber/NullAway/issues/983")
+  public void forEachLoop() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "  static @Nullable String[] fizz = {\"1\"};",
+            "  static void foo() {",
+            "    for (String s : fizz) {",
+            "      if (s != null) {",
+            "        s.toString();",
+            "      }",
+            "    }",
+            "    for (String s : fizz) {",
+            "      // BUG: Diagnostic contains: dereferenced expression s is @Nullable",
```

Comment:
Interesting... wonder how we are representing the access paths here. Might be a bit much to ask (basically, you need NullAway to remember "we checked all elements of this array and they are all `@NonNull`, therefore..."

Reply:
Ah, maybe we should clarify this test.  It's a bit confusing due to the multiple loops in the same method.  This was just meant to test that if the array contents are `@Nullable`, the loop variable should be treated as `@Nullable` (which it currently isn't; that why the test is ignored).  I wasn't thinking about using reasoning from the previous loop to avoid warning on the second warning.

@armughan11 when you work on #983, please rewrite this test to reorder the loops, to avoid this confusion